cmd/srcd/cmd/init_test.go

```diff
@@ -0,0 +1,205 @@
+	s.T().Helper()
+
+	ctx, cancel := context.WithTimeout(context.Background(), s.timeout)
+	defer cancel()
+
+	return s.RunSQL(ctx, "select 1")
+}
+
+func (s *InitTestSuite) getLogMessages(buf *bytes.Buffer) []string {
+	actualMsg := s.ParseLogMessages(buf)
+	var filteredMsg []string
+	for _, m := range actualMsg {
+		if m == "unable to list the available daemon versions on Docker Hub: Short version cannot contain PreRelease/Build meta data" {
+			continue
+		}
+
+		filteredMsg = append(filteredMsg, m)
+	}
+
+	return filteredMsg
+}
+
+func (s *InitTestSuite) TestWithoutWorkdir() {
+	require := s.Require()
+
+	buf, err := s.runInit("")
+	require.NoError(err)
+
+	actualMsg := s.getLogMessages(buf)
+	require.Equal(2, len(actualMsg))
+
+	workdir, _ := os.Getwd()
+	expectedMsg := [2]string{
+		fmt.Sprintf("starting daemon with working directory: %s", workdir),
+		"daemon started",
+	}
+
+	for i, exp := range expectedMsg {
+		require.Equal(exp, actualMsg[i])
+	}
+}
+
+func (s *InitTestSuite) TestWithValidWorkdir() {
+	require := s.Require()
+
+	buf, err := s.runInit(s.validWorkDir)
+	require.NoError(err)
+
+	actualMsg := s.getLogMessages(buf)
+	require.Equal(2, len(actualMsg))
+
+	expectedMsg := [2]string{
+		fmt.Sprintf("starting daemon with working directory: %s", s.validWorkDir),
+		"daemon started",
+	}
+
+	for i, exp := range expectedMsg {
+		require.Equal(exp, actualMsg[i])
+	}
+}
+
+func (s *InitTestSuite) TestWithInvalidWorkdir() {
+	require := s.Require()
+
+	buf, err := s.runInit(s.invalidWorkDir)
+	require.Error(err)
+
+	actualMsg := s.getLogMessages(buf)
+	require.Equal(1, len(actualMsg))
+
+	expectedMsg := [1]string{
+		fmt.Sprintf("path \\\"%s\\\" is not a valid working directory", s.invalidWorkDir),
+	}
+
+	for i, exp := range expectedMsg {
+		require.Equal(exp, actualMsg[i])
+	}
+}
+
+func (s *InitTestSuite) TestWithRunningDaemon() {
+	require := s.Require()
+
+	_, err := s.runInit(s.validWorkDir)
+	require.NoError(err)
+
+	buf, err := s.runInit(s.validWorkDir)
+	require.NoError(err)
+
+	actualMsg := s.getLogMessages(buf)
+	require.Equal(3, len(actualMsg))
+
+	expectedMsg := [3]string{
+		fmt.Sprintf("removing container %s", components.Daemon.Name),
+		fmt.Sprintf("starting daemon with working directory: %s", s.validWorkDir),
+		"daemon started",
+	}
+
+	for i, exp := range expectedMsg {
+		require.Equal(exp, actualMsg[i])
+	}
+}
+
+func (s *InitTestSuite) TestWithRunningOtherComponents() {
+	require := s.Require()
+
+	_, err := s.runInit(s.validWorkDir)
+	require.NoError(err)
+
+	_, err = s.runSQL()
+	require.NoError(err)
+
+	buf, err := s.runInit(s.validWorkDir)
+	require.NoError(err)
+
+	actualMsg := s.getLogMessages(buf)
+	require.Equal(5, len(actualMsg))
+
+	expectedMsg := [5]string{
+		fmt.Sprintf("removing container %s", components.Bblfshd.Name),
+		fmt.Sprintf("removing container %s", components.Daemon.Name),
+		fmt.Sprintf("removing container %s", components.Gitbase.Name),
```

Comment:
are those messages always get printed in the same order?

Reply:
Apparently yes, the test never failed. It depends on how docker returns the names of the running containers. But this could be improved for sure.